run_windows.go

```diff
@@ -0,0 +1,527 @@
+		// process is started, it does Chdir(attr.Dir). We are adjusting
+		// for that difference here by making argv0 absolute.
+		var err error
+		argv0, err = joinExeDirAndFName(attr.Dir, argv0)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+	argv0p, err := syscall.UTF16PtrFromString(argv0)
+	if err != nil {
+		return 0, 0, err
+	}
+
+	var cmdline string
+	// Windows CreateProcess takes the command line as a single string:
+	// use attr.CmdLine if set, else build the command line by escaping
+	// and joining each argument with spaces
+	if sys.CmdLine != "" {
+		cmdline = sys.CmdLine
+	} else {
+		cmdline = makeCmdLine(argv)
+	}
+
+	var argvp *uint16
+	if len(cmdline) != 0 {
+		argvp, err = syscall.UTF16PtrFromString(cmdline)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+
+	var dirp *uint16
+	if len(attr.Dir) != 0 {
+		dirp, err = syscall.UTF16PtrFromString(attr.Dir)
+		if err != nil {
+			return 0, 0, err
+		}
+	}
+
+	// Acquire the fork lock so that no other threads
+	// create new fds that are not yet close-on-exec
+	// before we fork.
+	syscall.ForkLock.Lock()
+	defer syscall.ForkLock.Unlock()
+
+	//p, _ := syscall.GetCurrentProcess()
+	//fd := make([]syscall.Handle, len(attr.Files))
+	//for i := range attr.Files {
+	//	if attr.Files[i] > 0 {
+	//		err := syscall.DuplicateHandle(p, syscall.Handle(attr.Files[i]), p, &fd[i], 0, true, syscall.DUPLICATE_SAME_ACCESS)
+	//		if err != nil {
+	//			return 0, 0, err
+	//		}
+	//		defer syscall.CloseHandle(syscall.Handle(fd[i]))
+	//	}
+	//}
+
+	// replaced default syscall.StartupInfo with custom startupInfEx for console handle
+	//si := new(syscall.StartupInfo)
+	//si.Cb = uint32(unsafe.Sizeof(*si))
+	si, err := createExtendedStartupInfo(consoleHandle)
+	if err != nil {
+		return 0, 0, err
+	}
+	// add finalizer for attribute list cleanup, best effort
+	runtime.SetFinalizer(si, func(si *startupInfoEx) {
+		_, _, _ = deleteProcThreadAttributeList.Call(uintptr(si.lpAttrList))
+	})
+
+	si.startupInfo.Flags = syscall.STARTF_USESTDHANDLES
+	if sys.HideWindow {
+		si.startupInfo.Flags |= syscall.STARTF_USESHOWWINDOW
+		si.startupInfo.ShowWindow = syscall.SW_HIDE
+	}
+	//si.StdInput = fd[0]
+	//si.StdOutput = fd[1]
+	//si.StdErr = fd[2]
+
+	pi := new(syscall.ProcessInformation)
+
+	flags := sys.CreationFlags | syscall.CREATE_UNICODE_ENVIRONMENT
+
+	// add startupInfoEx flag
+	flags = flags | windows.EXTENDED_STARTUPINFO_PRESENT
+
+	// ignore security attrs since both Process and Thread handles are not inheritable for conPty
+	if sys.Token != 0 {
+		err = syscall.CreateProcessAsUser(sys.Token, argv0p, argvp, nil, nil, false, flags, createEnvBlock(attr.Env), dirp, &si.startupInfo, pi)
+	} else {
+		err = syscall.CreateProcess(argv0p, argvp, nil, nil, false, flags, createEnvBlock(attr.Env), dirp, &si.startupInfo, pi)
+	}
+	if err != nil {
+		return 0, 0, err
+	}
+	defer syscall.CloseHandle(syscall.Handle(pi.Thread))
+
+	return int(pi.ProcessId), uintptr(pi.Process), nil
+}
+
+// copied from os/exec.Cmd
```

Comment:
It would be cleaner to create a `cmd_windows.go` and add some comments on why we need to do this.

Reply:
Agree, new files, and all its purpose is to implement slightly modified `syscallStartProces` function which is different from the one from the standard package, same reason as https://github.com/creack/pty/pull/109#discussion_r607878494

or we have to implement our own different exec.Cmd for Windows platform, which in my opinion is a sub-optimal solution as stated in the pr description.

Reply:
Updated comments for the empty assembly file, and moved cmd to cmd_windows.go